packages/grafana-ui/src/components/Menu/MenuItem.tsx

```diff
@@ -136,8 +140,15 @@ export const MenuItem = React.memo(
         aria-checked={ariaChecked}
         tabIndex={tabIndex}
       >
-        {icon && <Icon name={icon} className={styles.icon} aria-hidden />}
-        {label}
+        {renderCustomLabel ? (
+          ReactUtils.renderOrCallToRender(renderCustomLabel)
+        ) : (
+          <>
+            {icon && <Icon name={icon} className={styles.icon} aria-hidden />}
+            {label}
+          </>
+        )}
+
```

Comment:
I would try using label to support this customization use case (just extend the label type)

Reply:
yeah +1, i'd just switch label to be `React.ReactNode`. we'll need to do this anyway for translations since lingui passes `<Trans>` elements around instead of strings

Reply:
right. This could be a breaking change if we extend the support of label. I've browsed a little bit in our repo and `label` is being used as a `key` 🙈 . That is the reason I've use a new prop for this